app/models/agent.rb

```diff
@@ -61,8 +61,8 @@ class Agent < ActiveRecord::Base
   has_many :scenario_memberships, :dependent => :destroy, :inverse_of => :agent
   has_many :scenarios, :through => :scenario_memberships, :inverse_of => :agents
 
-  scope :active,   -> { where(disabled: false) }
-  scope :inactive, -> { where(disabled: true) }
+  scope :active,   -> { joins(:user).where('users.deactivated_at IS NULL').where(disabled: false) }
+  scope :inactive, -> { joins(:user).where(["users.deactivated_at IS NOT NULL OR disabled = ?", true]) }
```

Comment:
On my system it the query takes 0.1ms instead of 0.05ms with about 100 agents, so i guess it will be hardly noticeable, but better safe then sorry :smile: 
@TildeWill disabled users can not create agents because they cannot login (even active sessions will be revoked by devise)